### PR TITLE
Vsts git support

### PR DESCRIPTION
**Pull request by dpmerron-ltd**
I have added support for legacy Azure DevOps git accounts that have been migrated from VSTS to Azure DevOps. 

Instead of the host `ssh.dev.azure.com` it uses `vs-ssh.visualstudio.com`.

Updated documentation to reflect these changes also. 

I have been unable to run tests as I can't seem to get the code to compile - errors finding ./fluxctl so wondered if someone else were able to pick up the testing part on my behalf? Happy to spend more time to do it but right now have other important things I need to work on.

Any feedback, or suggestions appreciated.

### CODE REVIEW

**Comment by squaremo** [comment]
Looks reasonable, thank you @dpmerron-ltd. One thing: can you mention the method used to obtain the fingerprint, in the comment in verify-known-hosts.sh -- ta.

File: site/faq.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ menu_order: 60
   * [How often does Flux check for new images?](#how-often-does-flux-check-for-new-images)
   * [How often does Flux check for new git commits (and can I make it sync faster)?](#how-often-does-flux-check-for-new-git-commits-and-can-i-make-it-sync-faster)
   * [How do I use my own deploy key?](#how-do-i-use-my-own-deploy-key)
-  * [How do I use a private git host (or one that's not github.com, gitlab.com, bitbucket.org, or dev.azure.com)?](#how-do-i-use-a-private-git-host-or-one-thats-not-githubcom-gitlabcom-bitbucketorg-or-devazurecom)
+  * [How do I use a private git host (or one that's not github.com, gitlab.com, bitbucket.org, dev.azure.com, or vs-ssh.visualstudio.com)?](#how-do-i-use-a-private-git-host-or-one-thats-not-githubcom-gitlabcom-bitbucketorg-or-devazurecom)
```
Comment (hiddeco) [inline site/faq.md]:
```suggestion
  * [How do I use a private git host (or one that's not github.com, gitlab.com, bitbucket.org, dev.azure.com, or vs-ssh.visualstudio.com)?](#how-do-i-use-a-private-git-host-or-one-thats-not-githubcom-gitlabcom-bitbucketorg-devazurecom-or-vs-sshvisualstudiocom)
```

**Comment by dpmerron-ltd** [comment]
> Looks reasonable, thank you @dpmerron-ltd. One thing: can you mention the method used to obtain the fingerprint, in the comment in verify-known-hosts.sh -- ta.

@squaremo I have added that comment - just added to the existing Azure DevOps comment with an `&` as the process is exactly the same. 

Dan

File: docker/verify_known_hosts.sh [hunk 2]
```diff
@@ -17,8 +17,8 @@ known_hosts_file=${known_hosts_file:-/etc/ssh/ssh_known_hosts}
 #  - github.com: https://help.github.com/articles/github-s-ssh-key-fingerprints/
 #  - gitlab.com: https://docs.gitlab.com/ee/user/gitlab_com/#ssh-host-keys-fingerprints
 #  - bitbucket.org: https://confluence.atlassian.com/bitbucket/ssh-keys-935365775.html
-#  - ssh.dev.azure.com: sign in, then go to User settings -> SSH Public Keys
-#    (this is where the public key fingerprint is shown; it's not a setting)
+#  - ssh.dev.azure.com & vs-ssh.visualstudio.com: sign in, then go to User settings -> SSH Public Keys
+#    (this is where the public key fingerprint is shown; it's not a setting)s
```
Comment (squaremo) [inline docker/verify_known_hosts.sh]:
```suggestion
#    (this is where the public key fingerprint is shown; it's not a setting)
```

**Comment by squaremo** [comment]
> just added to the existing Azure DevOps comment

Great, thanks for that -- I also found somewhere it's actually published https://docs.microsoft.com/en-us/azure/devops/repos/git/use-ssh-keys-to-authenticate?view=azure-devops

Sorry for this last bit of busywork: can you squash it all into one commit, please (we've got up to seven, with the github suggestions, etc.)

**Comment by squaremo** [comment]
Thanks @dpmerron-ltd! (and @hiddeco for the review)

**Comment by dpmerron-ltd** [comment]
Thanks!

**Pull request merged**